### PR TITLE
`Development`: Fix issues with server test flakiness

### PR DESCRIPTION
**Pull request by ole-ve**
### Motivation and Context
Some server tests are flaky after changing the test order by restructuring into modules.

### Description
Following improvements:
- Clear the pending participant scores calculations between each test
- Enable the participant score scheduler service for each test
- Make assertions and `mockito-verify`'s not depend on an absolut state
- Ignore garbage collector file `gc.log.lock` generated by JGit in each repository on zipping and deletion
- Add own test prefix for some LocalCILocalVC-tests that re-using the same prefix
- Increase timeout for some specific tests

### Out-of-scope
- Find the issue within the LocalCILocalVC implementation that causes the first test of `LocalVCLocalCIIntegrationTest` to fail (no matter of the order)
- Override `Object#equals()` in each atlas entity/domain object to ensure that `Mockito.verify` does not depend on the entity/domain object being in memory (and not "unproxied" or whatever might happen in the background)

## Summary by CodeRabbit

- **New Features**
	- Enhanced file zipping process to exclude specific files, improving efficiency.
	- Improved notification testing framework with new constants and verification logic for tutorial group management.

- **Bug Fixes**
	- Updated equality checks for `expirationDate` to ensure consistent behavior across date types.

- **Tests**
	- Streamlined test setup and teardown processes for various integration tests, enhancing clarity and reducing complexity.
	- Enhanced integration tests for student participation management in exams, ensuring accurate deletion of participations and submissions.

- **Chores**
	- Removed unused imports and unnecessary method calls in tests to simplify codebase.

### CODE REVIEW

File: src/main/java/de/tum/cit/aet/artemis/atlas/domain/competency/CourseCompetency.java [hunk 1]
```diff
@@ -236,4 +236,6 @@ public void setLearningPaths(Set<LearningPath> learningPaths) {
     public void prePersistOrUpdate() {
         this.lectureUnits.removeIf(lectureUnit -> lectureUnit instanceof ExerciseUnit);
     }
+
+    // ToDo: Implement equals to ensure that mockito verify works correctly
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/cit/aet/artemis/atlas/domain/competency/CourseCompetency.java:240]: Why isn't the equals from `DomainObject` enough?

File: src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java [hunk 2]
```diff
@@ -170,10 +179,18 @@ private SshClient clientConnectToArtemisSshServer() throws GeneralSecurityExcept
 
         serverSessions = sshServer.getActiveSessions();
         assertThat(clientSession.isAuthenticated()).isTrue();
-        assertThat(serverSessions.size()).isEqualTo(numberOfSessions + 1);
+        assertThat(serverSessions.size()).as("There are more server sessions activated than expected.").isEqualTo(numberOfSessions + 1);
```
Comment (JohannesStoehr) [inline src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java:182]: This test seams to fail now

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Looks good overall, left two comments:

File: src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java (hunk 2 above)
Comment (ole-ve) [reply src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java:182]: It's still flaky, but just a _little_ bit more reliable.
I honestly don't really know how to improve this. I can only imagine that the underlying SSHD library might be opening up some additional sessions but I think it's somehow the task of the people maintaining this :)

Comment (krusche) [reply src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java:182]: @BBesrour @SimonEntholzer can you please have a look?

**Review by JohannesStoehr — APPROVED** [review]

File: src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java (hunk 2 above)
Comment (SimonEntholzer) [reply src/test/java/de/tum/cit/aet/artemis/programming/localvcci/LocalVCSshIntegrationTest.java]: I will fix that in a small followup

**Review by SimonEntholzer — APPROVED** [review]
Code

**Pull request merged**